File: backend/src/main/java/com/laras/service/ContactMessageService.java
package com.laras.service;

import com.laras.dto.ContactMessageDto;
import com.laras.entity.ContactMessage;
import com.laras.repository.ContactMessageRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
@Transactional
public class ContactMessageService {

    private final ContactMessageRepository contactMessageRepository;

    public List<ContactMessageDto> getAllMessages() {
        return contactMessageRepository.findAllByOrderByCreatedAtDesc()
                .stream()
                .map(this::toDto)
                .collect(Collectors.toList());
    }

    public List<ContactMessageDto> getUnreadMessages() {
        return contactMessageRepository.findByIsReadFalseOrderByCreatedAtDesc()
                .stream()
                .map(this::toDto)
                .collect(Collectors.toList());
    }

    public long getUnreadCount() {
        return contactMessageRepository.countByIsReadFalse();
    }

    public ContactMessageDto getMessageById(Long id) {
        ContactMessage message = contactMessageRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Message not found with id: " + id));
        return toDto(message);
    }

    public ContactMessageDto createMessage(ContactMessageDto dto) {
        ContactMessage message = ContactMessage.builder()
                .name(dto.getName())
                .email(dto.getEmail())
                .phone(dto.getPhone())
                .message(dto.getMessage())
                .isRead(false)
                .build();

        ContactMessage saved = contactMessageRepository.save(message);
        return toDto(saved);
    }

    public void markAsRead(Long id) {
        ContactMessage message = contactMessageRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Message not found with id: " + id));
        message.setIsRead(true);
        contactMessageRepository.save(message);
    }

    public void markAsUnread(Long id) {
        ContactMessage message = contactMessageRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Message not found with id: " + id));
        message.setIsRead(false);
        contactMessageRepository.save(message);
    }

    public void deleteMessage(Long id) {
        contactMessageRepository.deleteById(id);
    }

    private ContactMessageDto toDto(ContactMessage message) {
        return ContactMessageDto.builder()
                .id(message.getId())
                .name(message.getName())
                .email(message.getEmail())
                .phone(message.getPhone())
                .message(message.getMessage())
                .isRead(message.getIsRead())
                .createdAt(message.getCreatedAt())
                .updatedAt(message.getUpdatedAt())
                .build();
    }
}

File: frontend/src/services/contactMessageService.ts
import api from './api';

export interface ContactMessage {
    id: number;
    name: string;
    email: string;
    phone?: string;
    message: string;
    isRead: boolean;
    createdAt: string;
    updatedAt: string;
}

export interface ContactMessageInput {
    name: string;
    email: string;
    phone?: string;
    message: string;
}

export const contactMessageService = {
    submitContactForm: async (data: ContactMessageInput): Promise<ContactMessage> => {
        const response = await api.post<ContactMessage>('/contact', data);
        return response.data;
    },

    getAllMessages: async (): Promise<ContactMessage[]> => {
        const response = await api.get<ContactMessage[]>('/admin/messages');
        return response.data;
    },

    getUnreadMessages: async (): Promise<ContactMessage[]> => {
        const response = await api.get<ContactMessage[]>('/admin/messages/unread');
        return response.data;
    },

    getUnreadCount: async (): Promise<number> => {
        const response = await api.get<{ unread: number }>('/admin/messages/count');
        return response.data.unread;
    },

    markAsRead: async (id: number): Promise<void> => {
        await api.patch(`/admin/messages/${id}/read`);
    },

    markAsUnread: async (id: number): Promise<void> => {
        await api.patch(`/admin/messages/${id}/unread`);
    },

    deleteMessage: async (id: number): Promise<void> => {
        await api.delete(`/admin/messages/${id}`);
    },
};

export default contactMessageService;
